backend/src/main/java/org/cryptomator/hub/api/DeviceResource.java

```diff
@@ -178,16 +219,29 @@ public record DeviceDto(@JsonProperty("id") @ValidId String id,
 							@JsonProperty("owner") @ValidId String ownerId,
 							@JsonProperty("creationTime") Instant creationTime,
 							@JsonProperty("lastIpAddress") String lastIpAddress,
-							@JsonProperty("lastAccessTime") Instant lastAccessTime) {
+							@JsonProperty("lastAccessTime") Instant lastAccessTime,
+							@JsonProperty("legacyDevice") boolean legacyDevice) {
 
 		public static DeviceDto fromEntity(Device entity) {
-			return new DeviceDto(entity.getId(), entity.getName(), entity.getType(), entity.getPublickey(), entity.getUserPrivateKeys(), entity.getOwner().getId(), entity.getCreationTime().truncatedTo(ChronoUnit.MILLIS), null, null);
+			return new DeviceDto(entity.getId(), entity.getName(), entity.getType(), entity.getPublickey(), entity.getUserPrivateKeys(), entity.getOwner().getId(), entity.getCreationTime().truncatedTo(ChronoUnit.MILLIS), null, null, false);
+		}
+
+		@Deprecated
+		public static DeviceDto fromEntity(LegacyDevice entity) {
+			return new DeviceDto(entity.getId(), entity.getName(), entity.getType(), null, null, entity.getOwner().getId(), entity.getCreationTime().truncatedTo(ChronoUnit.MILLIS), null, null, true);
```

Comment:
Edd `entity.getPublicKey()` to fulfil `@NotNull`. It doesn't exist in the entity class yet, but is present in the DB.

Reply:
Fixed in 00bdeb1